packages/theme/modules/customer/components/LoginModal/LoginModal.vue

```diff
@@ -0,0 +1,251 @@
+          :show-recaptcha="isRecaptchaEnabled"
+          :loading="loading"
+          :form="registerForm"
+          @submit="form => onRegisterFormSubmit(form)"
+          @go-to-login="currentlyDisplayedForm = 'login'"
+        >
+          <template #error>
+            {{ userError.register && userError.register.message }}
+          </template>
+        </RegisterForm>
+        <ForgotPasswordForm
+          v-else-if="currentlyDisplayedForm === 'forgotPassword'"
+          data-testid="forgot-password-form"
+          :loading="forgotPasswordLoading"
+          :show-recaptcha="isRecaptchaEnabled"
+          :form="forgotPasswordForm"
+          @submit="form => onForgotPasswordFormSubmit(form)"
+        >
+          <template #error>
+            {{ forgotPasswordError.request && forgotPasswordError.request.message }}
+          </template>
+        </ForgotPasswordForm>
+        <ForgotPasswordThankYou
+          v-else-if="currentlyDisplayedForm === 'forgotPasswordThankYou'"
+          data-testid="forgot-password-thank-you"
+          :form="forgotPasswordForm"
+        >
+          <template #email>
+            {{ forgotPasswordForm.username }}
+          </template>
+        </ForgotPasswordThankYou>
+      </div>
+    </transition>
+  </SfModal>
+</template>
+<script lang="ts">
+import {
+  ref,
+  defineComponent,
+  computed,
+  useContext,
+} from '@nuxtjs/composition-api';
+import {
+  SfModal,
+  SfBar,
+} from '@storefront-ui/vue';
+import { extend } from 'vee-validate';
+import { required, email } from 'vee-validate/dist/rules';
+import { useUiState } from '~/composables/useUiState';
+import useCart from '~/modules/checkout/composables/useCart';
+import useWishlist from '~/modules/wishlist/composables/useWishlist';
+import { useForgotPassword } from '~/modules/customer/composables/useForgotPassword';
+import { useUser } from '~/modules/customer/composables/useUser';
+import { customerPasswordRegExp, invalidPasswordMsg } from '~/modules/customer/helpers/passwordValidation';
+
+import LoginForm from './forms/LoginForm.vue';
+import RegisterForm from './forms/RegisterForm.vue';
+import ForgotPasswordForm from './forms/ForgotPasswordForm.vue';
+import ForgotPasswordThankYou from './forms/ForgotPasswordThankYou.vue';
+import { ForgotPasswordFormFields, LoginFormFields, RegisterFormFields } from './forms/types';
+
+extend('email', {
+  ...email,
+  message: 'Invalid email',
+});
+
+extend('required', {
+  ...required,
+  message: 'This field is required',
+});
+
+extend('password', {
+  message: invalidPasswordMsg,
+  validate: (value: string) => customerPasswordRegExp.test(value),
+});
+
+type FormName = 'login' | 'register' | 'forgotPassword' | 'forgotPasswordThankYou';
+
+export default defineComponent({
+  name: 'LoginModal',
+  components: {
+    SfModal,
+    SfBar,
+
+    LoginForm,
+    RegisterForm,
+    ForgotPasswordForm,
+    ForgotPasswordThankYou,
+  },
+  setup(_props, { emit }) {
```

Comment:
why props are underscored? if you are not using props use just `_`
```suggestion
  setup(_, { emit }) {
```

Reply:
I don't think this is important as there's no difference, but I changed it in c3e6870daaea409e8d30683a721b59feaf8e103f because it's quicker than arguing

Reply:
Probably, but this is a common approach in many docs, blogs, etc.